kvserver: TestStoreRangeMergeTimestampCache is flaky and kills the whole package - fatal on goroutine

Take a look at the test failure below. There's a failure and then a panic; I think the panic comes from a `t.Fatal` run in a filter, on a random goroutine: https://github.com/cockroachdb/cockroach/blob/0ba5a73de8ed327a1d7b92408b7d3ac466d54656/pkg/kv/kvserver/client_merge_test.go#L677

https://teamcity.cockroachdb.com/repository/download/Cockroach_UnitTests_Testrace/3070524:id/raw.2.json.txt.tgz!/artifacts/full_output.txt

<details>
<summary>Test output:</summary>
=== RUN   TestStoreRangeMergeTimestampCache/disjoint-leaseholders=true/through-snapshot=true/future-read=false
    client_merge_test.go:742: condition failed to evaluate within 45s: merge not applied
        goroutine 131373 [running]:
        runtime/debug.Stack(0xc003f2ac50, 0x7e40900, 0xc0078832e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7eff080, 0xc004f11680, 0xc003f2ac50)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x85
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.mergeCheckingTimestampCaches(0xc004f11680, 0x101)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:742 +0x23dd
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeMergeTimestampCache.func1.1.1(0xc004f11680, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:405 +0x5c
        github.com/cockroachdb/cockroach/pkg/testutils.RunTrueAndFalse.func1(0xc004f11680)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:23 +0x6f
        testing.tRunner(0xc004f11680, 0xc005693920)
        	/usr/local/go/src/testing/testing.go:1123 +0x203
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x5bc
E210610 22:28:48.106766 134458 kv/kvserver/queue.go:1093  [n3,replicaGC,s3,r38/3:{b-/Table/Systeâ€¦}] 7  failed to send RPC: sending to all replicas failed; last error: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:41672->127.0.0.1:42031: use of closed network connection
E210610 22:28:48.118833 134345 kv/kvserver/queue.go:1093  [n2,replicaGC,s2,r38/2:{b-/Table/Systeâ€¦}] 8  failed to send RPC: sending to all replicas failed; last error: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:41660->127.0.0.1:42031: use of closed network connection
    client_merge_test.go:677: condition failed to evaluate within 45s: expected [0 4 4], got [0 0 0]
        goroutine 134078 [running]:
        runtime/debug.Stack(0xc008a420b0, 0x7e40900, 0xc007388420)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x7eff080, 0xc004f11680, 0xc008a420b0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x85
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).WaitForValues(0xc0047dd200, 0x80235e0, 0xc004f11680, 0xc00631e088, 0x1, 0x1, 0xc000972ce0, 0x3, 0x3)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1316 +0x110
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.mergeCheckingTimestampCaches.func8()
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:677 +0x7ea
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.mergeCheckingTimestampCaches.func1(0x7ed2e40, 0xc001ef73e0, 0x16875908512aac00, 0x2, 0x100000001, 0x1, 0x0, 0x5, 0x0, 0xc003d7fc20, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:433 +0x1b2
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID(0xc005f76000, 0x7ed2e40, 0xc001ef73b0, 0x5, 0xc006a06580, 0x2, 0x1)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:93 +0x8c2
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:43
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send(0xc0018b5800, 0x7ed2e40, 0xc001ef7380, 0x16875908512aac00, 0x2, 0x100000001, 0x1, 0x0, 0x5, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:196 +0x537
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).Send(0xc00663c880, 0x7ed2e40, 0xc001ef7380, 0x0, 0x0, 0x100000001, 0x1, 0x0, 0x5, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:191 +0x17f
        github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal.func1(0x7ed2e40, 0xc001ef7380, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:890 +0x27e
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0063ef9e0, 0x7ed2e40, 0xc001ef7380, 0x64ca750, 0x10, 0xc008a43858, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:326 +0x12a
        github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc00515a000, 0x7ed2e40, 0xc001ef7380, 0xc006a064d0, 0x0, 0x0, 0x1)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:878 +0x267
        github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc00515a000, 0x7ed2e40, 0xc001ef7350, 0xc006a064d0, 0x5d899c, 0xc001ef7290, 0x4a246c)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:945 +0x106
        github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch(...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:462
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch(0xc00284d640, 0x7ed2e40, 0xc001ef7350, 0x1, 0x7f1f360, 0xc00f353d50, 0x0, 0x0, 0x100000001, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:193 +0x25d
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext(0xc00284d640, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x100000001, 0x1, 0x0, 0x5, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:175 +0x25c
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.raceTransport.SendNext(0x7f28060, 0xc00284d640, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport_race.go:78 +0x443
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas(0xc003716c00, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1860 +0xe55
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch(0xc003716c00, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1478 +0x410
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges(0xc003716c00, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1116 +0x1fbf
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send(0xc003716c00, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:755 +0x75b
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnLockGatekeeper).SendLocked(0xc0054aa460, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_lock_gatekeeper.go:82 +0x1b5
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnMetricRecorder).SendLocked(0xc0054aa428, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_metric_recorder.go:46 +0xfb
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnCommitter).SendLocked(0xc0054aa3f8, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_committer.go:195 +0x1f5
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).sendLockedWithRefreshAttempts(0xc0054aa358, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:240 +0x115
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).SendLocked(0xc0054aa358, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:175 +0x386
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnPipeliner).SendLocked(0xc0054aa288, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go:255 +0x295
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSeqNumAllocator).SendLocked(0xc0054aa268, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_seq_num_allocator.go:105 +0x355
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnHeartbeater).SendLocked(0xc0054aa1c8, 0x7ed2e40, 0xc001ef7290, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:241 +0x275
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*TxnCoordSender).Send(0xc0054aa000, 0x7ed2e40, 0xc001e04f60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_coord_sender.go:513 +0x575
        github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender(0xc00663c480, 0x7ed2e40, 0xc001e04f60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:817 +0x162
        github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Send(0xc0082b5ce0, 0x7ed2e40, 0xc001e04f60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:982 +0x228
        github.com/cockroachdb/cockroach/pkg/kv.sendAndFill(0x7ed2e40, 0xc001e04f60, 0xc008a4ba70, 0xc00503ef00, 0x1, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:728 +0x1e3
        github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run(0xc0082b5ce0, 0x7ed2e40, 0xc001e04f60, 0xc00503ef00, 0xc001612000, 0x3)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:635 +0xd9
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).AdminMerge.func1(0xc0082b5ce0, 0xc001e04f60, 0xc00663c480)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:764 +0x1c70
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).AdminMerge(0xc005f76000, 0x7ed2e40, 0xc001e04f60, 0xc000a78ee8, 0x4, 0x8, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:785 +0x1e5
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeAdminBatch(0xc005f76000, 0x7ed2e40, 0xc001e04f60, 0xc0082b5c30, 0x100000001, 0x1)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:699 +0x13c9
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID(0xc005f76000, 0x7ed2e40, 0xc00b811da0, 0x5, 0xc0082b5c30, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:110 +0x705
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:43
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send(0xc0018b5800, 0x7ed2dc0, 0xc0001a6010, 0x16875908512aac00, 0x0, 0x100000001, 0x1, 0x0, 0x5, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:196 +0x537
        github.com/cockroachdb/cockroach/pkg/kv.Wrap.func1(0x7ed2dc0, 0xc0001a6010, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/sender.go:456 +0x13b
        github.com/cockroachdb/cockroach/pkg/kv.SenderFunc.Send(0xc008dc6a80, 0x7ed2dc0, 0xc0001a6010, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/sender.go:392 +0xc2
        github.com/cockroachdb/cockroach/pkg/kv.SendWrappedWith(0x7ed2dc0, 0xc0001a6010, 0x7e433a0, 0xc008dc6a80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/sender.go:432 +0x17b
        github.com/cockroachdb/cockroach/pkg/kv.SendWrapped(...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/sender.go:449
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.mergeCheckingTimestampCaches.func10(0xc004994b00, 0x7ed2dc0, 0xc0001a6010, 0xc0018b5800, 0xc002655080)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:737 +0x19a
        created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.mergeCheckingTimestampCaches
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:735 +0x2345
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2c9efe6]

</details>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -411,19 +411,22 @@ func TestStoreRangeMergeTimestampCache(t *testing.T) {
 func mergeCheckingTimestampCaches(
 	t *testing.T, disjointLeaseholders, throughSnapshot, futureRead bool,
 ) {
-	// mergeCommitFilter is used to issue a sequence of operations on the LHS of
-	// a range merge immediately before it.
-	var mergeCommitFilter func()
-	// blockHBAndGCs is used to black hole Heartbeat and GC requests for the
-	// duration of the merge on the throughSnapshot path. Neither request type
-	// is needed and both can create issues by holding latches during the split
-	// leader-leaseholder state.
-	var blockHBAndGCs chan struct{}
-	var filterMu syncutil.Mutex
+
+	var filterMu struct {
+		syncutil.Mutex
+		// mergeCommitFilter is used to issue a sequence of operations on the LHS of
+		// a range merge immediately before it.
+		mergeCommitFilter func()
+		// blockHBAndGCs is used to black hole Heartbeat and GC requests for the
+		// duration of the merge on the throughSnapshot path. Neither request type
+		// is needed and both can create issues by holding latches during the split
+		// leader-leaseholder state.
+		blockHBAndGCs chan struct{}
+	}
 	testingRequestFilter := func(_ context.Context, ba roachpb.BatchRequest) *roachpb.Error {
 		filterMu.Lock()
-		mergeCommitFilterCopy := mergeCommitFilter
-		blockHBAndGCsCopy := blockHBAndGCs
+		mergeCommitFilterCopy := filterMu.mergeCommitFilter
+		blockHBAndGCsCopy := filterMu.blockHBAndGCs
 		filterMu.Unlock()
 		for _, req := range ba.Requests {
 			switch v := req.GetInner().(type) {
@@ -476,6 +479,17 @@ func mergeCheckingTimestampCaches(
 			},
 		})
 	defer tc.Stopper().Stop(ctx)
+	defer func() {
+		filterMu.Lock()
+		defer filterMu.Unlock()
+		if filterMu.blockHBAndGCs != nil {
+			// If test failed before closing the channel, do so now
+			// to (maybe) avoid hangs. Note that this must execute
+			// before the Stopper().Stop() call above.
+			close(filterMu.blockHBAndGCs)
+		}
+	}()
+
 	lhsStore := tc.GetFirstStoreFromServer(t, 0)
 	var rhsStore *kvserver.Store
 	if disjointLeaseholders {
@@ -642,7 +656,7 @@ func mergeCheckingTimestampCaches(
 		snapChan := make(chan kvserver.IncomingSnapshot, 1)
 
 		filterMu.Lock()
-		mergeCommitFilter = func() {
+		filterMu.mergeCommitFilter = func() {
 			// Install leader-leaseholder partition.
 			for i, s := range lhsStores {
 				var funcs unreliableRaftHandlerFuncs
@@ -720,7 +734,7 @@ func mergeCheckingTimestampCaches(
 		// Begin blocking txn heartbeats and GC requests. They cause issues
 		// because they can grab latches and then get stuck once in the split
 		// leader-leaseholder state.
-		blockHBAndGCs = make(chan struct{})
+		filterMu.blockHBAndGCs = make(chan struct{})
 
 		// Install a filter to capture the Raft snapshot.
 		snapshotFilter = func(inSnap kvserver.IncomingSnapshot) {
@@ -764,15 +778,24 @@ func mergeCheckingTimestampCaches(
 			}
 			tc.Servers[i].RaftTransport().Listen(s.StoreID(), h)
 		}
-		close(blockHBAndGCs)
+		close(filterMu.blockHBAndGCs)
+		filterMu.Lock()
+		filterMu.blockHBAndGCs = nil
+		filterMu.Unlock()
 
 		t.Logf("waiting for snapshot to LHS leaseholder")
-		inSnap := <-snapChan
+		var inSnap kvserver.IncomingSnapshot
+		select {
+		case inSnap = <-snapChan:
+		case <-time.After(45 * time.Second):
+			t.Fatal("timed out waiting for snapChan")
+		}
 		inSnapDesc := inSnap.State.Desc
 		require.Equal(t, lhsDesc.StartKey, inSnapDesc.StartKey)
 		require.Equal(t, rhsDesc.EndKey, inSnapDesc.EndKey)
 
 		// Wait for all async ops to complete.
+		after45s := time.After(45 * time.Second)
 		for _, asyncRes := range []struct {
 			name string
 			ch   chan *roachpb.Error
@@ -782,7 +805,12 @@ func mergeCheckingTimestampCaches(
 			{"merge", mergeChan},
 		} {
 			t.Logf("waiting for result of %s", asyncRes.name)
-			err := <-asyncRes.ch
+			var err *roachpb.Error
+			select {
+			case err = <-asyncRes.ch:
+			case <-after45s:
+				t.Fatalf("timed out on %s", asyncRes.name)
+			}
 			require.NotNil(t, err, "%s should fail", asyncRes.name)
 			require.Regexp(t, "result is ambiguous", err, "%s's result should be ambiguous", asyncRes.name)
 		}

